rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -7,145 +7,120 @@
+      def base_path_from_servers(openapi_spec, use_server = :default)
+        return '' if openapi_spec[:servers].nil? || openapi_spec[:servers].empty?
+        server = openapi_spec[:servers].first
         variables = {}
         server.fetch(:variables, {}).each_pair { |k,v| variables[k] = v[use_server] }
         base_path = server[:url].gsub(/\{(.*?)\}/) { variables[$1.to_sym] }
         URI(base_path).path
       end
 
-      def add_path(request, metadata, swagger_doc, parameters, example)
-        open_api_3_doc = doc_version(swagger_doc).start_with?('3')
-        uses_base_path = swagger_doc[:basePath].present?
-
-        if open_api_3_doc && uses_base_path
-          Rswag::Specs.deprecator.warn('Rswag::Specs: WARNING: basePath is replaced in OpenAPI3! Update your swagger_helper.rb')
-        end
-
-        if uses_base_path
-          template = (swagger_doc[:basePath] || '') + metadata[:path_item][:template]
-        else # OpenAPI 3
-          template = base_path_from_servers(swagger_doc) + metadata[:path_item][:template]
-        end
+      def add_path(request, metadata, openapi_spec, parameters, example)
+        template = base_path_from_servers(openapi_spec) + metadata[:path_item][:template]
 
         request[:path] = template.tap do |path_template|
           parameters.select { |p| p[:in] == :path }.each do |p|
-            unless example.respond_to?(extract_getter(p))
-              raise ArgumentError.new("`#{p[:name].to_s}` parameter key present, but not defined within example group"\
+            begin
+              param_value = params.fetch(p[:name].to_s).to_s
+            rescue KeyError
+              raise ArgumentError.new("`#{p[:name]}`" \
+                "parameter key present, but not defined within example group" \
                 "(i. e `it` or `let` block)")
             end
-            path_template.gsub!("{#{p[:name]}}", example.send(extract_getter(p)).to_s)
+            path_template.gsub!("{#{p[:name]}}", param_value)
           end
 
-          parameters.select { |p| p[:in] == :query }.each_with_index do |p, i|
+          parameters.select { |p| p[:in] == :query && params.key?(p[:name]) }.each_with_index do |p, i|
             path_template.concat(i.zero? ? '?' : '&')
-            path_template.concat(build_query_string_part(p, example.send(extract_getter(p)), swagger_doc))
+            path_template.concat(build_query_string_part(p, params.fetch(p[:name]), openapi_spec))
           end
         end
       end
 
-      def build_query_string_part(param, value, swagger_doc)
+      def build_query_string_part(param, value, openapi_spec)
         name = param[:name]
         escaped_name = CGI.escape(name.to_s)
 
-        # OAS 3: https://swagger.io/docs/specification/serialization/
-        if swagger_doc && doc_version(swagger_doc).start_with?('3') && param[:schema]
+        # NOTE: https://swagger.io/docs/specification/serialization/
+        if param[:schema]
```

Comment:
I can't add a comment down there, but shouldn't the `else` case be removed? Since it relies on `type` and `collectionFormat` which just do not exist anymore in the new OpenAPI version.

Reply:
I also saw that RSwag explicitly supports using `type: ...` instead of the "correct" version`schema: { type: ... }`.  (in [this commit](https://github.com/rswag/rswag/commit/70eb277e049c95410aafb77807e227e4c85faebe)).

This actually leads to some confusing behaviour because in the current RSwag version: the only place where using `type: ` does not work is when trying to implement a specific serialization strategy (because then there is no `schema` key of course).

So I would move the
```
type = type = param[:type] || param.dig(:schema, :type)
```
to the top of the file instead, remove this condition altogether and then use that type for serialization.

Reply:
> https://github.com/rswag/rswag/pull/714/files#r1439746426

It can be removed. Thanks for pointing it out!

Reply:
> https://github.com/rswag/rswag/pull/714/files#r1439783378

I see `param[:schema][:type]` being handled in the case above:
https://github.com/rswag/rswag/blob/dc9c8bc894074295abad6afadfa4f0a8772c8b82/rswag-specs/lib/rswag/specs/request_factory.rb#L139

Correct me if I'm wrong, but it seems to me that the `type = type = param[:type] || param.dig(:schema, :type)` can be just removed

Reply:
Reminder to handle invalid attributes:

https://github.com/rswag/rswag/commit/812d311426c323c76d32ee7fdb17931d7f98f565#r148058990